src/pocketmine/updater/AutoUpdater.php

```diff
@@ -59,7 +59,12 @@ public function __construct(Server $server, $endpoint){
 	 */
 	public function checkUpdateCallback(array $updateInfo){
 		$this->updateInfo = $updateInfo;
-		$this->checkUpdate();
+		if(!$this->checkUpdate()){
+			if($this->server->getProperty("auto-updater.on-update.warn-console", true)){
+				$this->server->getLogger("[AutoUpdater] Receivied invalid response, failed to check for update.");
```

Comment:
> f an invalid response is given console will be warned

Really? That's an amazing log message you have there... one that the console will never see...